airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionPageTitle.tsx

```diff
@@ -57,11 +59,13 @@ const ConnectionPageTitle: React.FC<IProps> = ({ source, destination, currentSte
       id: ConnectionSettingsRoutes.TRANSFORMATION,
       name: <FormattedMessage id={"connectionForm.transformation.title"} />,
     },
-    {
+  ];
+
+  connection.status !== ConnectionStatus.DEPRECATED &&
```

Comment:
Similar logic needs to be applied to the route itself to avoid users navigating to the settings from a URL.